src/intrinsics/prepack/global.js

```diff
@@ -138,14 +139,17 @@ export default function(realm: Realm): void {
         "global.__registerReactComponentRoot",
         "__registerReactComponentRoot",
         0,
-        (context, [functionValue]) => {
-          invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
+        (context, [value]) => {
+          invariant(
```

Comment:
Prefer some sort of CompilerDiagnostic here to an invariant because this is a user-callable function.